### ISSUE TITLE
SA1118 doesn't allow to write anonymous object (or object with property initialization) on multiple lines.

### ISSUE THREAD

**Issue by mpawelski** [DotNetAnalyzers/StyleCopAnalyzers#2048]
This code:

``` csharp
          Test(
                "as",
                new
                {
                    foo = 1, bar = 2
                });
```

and this:

``` csharp
          Test(
                "as",
                new MyObject
                {
                    Foo = 1
                });
```

 for this method:

``` csharp
        public void Test(string test, object obj)
        {
        }
```

 gives SA1118 warnig when in StyleCop Classic it didn't. I'm migrating from classic StyleCop and had to remove this rule because of that. I think it should be fixed.

BTW Thanks for working on this projects! It's great that we can use StyleCop rules and use C#6 :)

**Comment by sharwell** [comment]
:memo: According to the [StyleCop Classic documentation](http://www.stylecop.com/docs/SA1118.html) for this rule, our current implementation is behaving as expected.

Depending on how discussion here goes, this behavior should either be changed and documented, or we should update the [Known Changes](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/KnownChanges.md) documentation to include this item

**Comment by mpawelski** [comment]
Yes, it looks that current implementation matches documentation, but if we are going to leave it as it is I really think it should not be enabled by default (also discussed in #823). I don't think current implementation is that usable and I actually think that it decrease readability.

I see that it was also discussed in #1597. So what actually StyleCop Classic implementation checks? Only if parameter is splitted by "+" and span multiple lines?.

**Comment by vweijsters** [comment]
I think both cases above can be considered in line with the changes made for #1597.
I would therefore propose to change the current implementation to accept both cases (ObjectCreationExpression and AnonymousObjectCreationExpression).

**Comment by paulomorgado** [comment]
The rule was created before the feature was added to the language and doesn't make sense if taken literally.

I think the correct implementation should be taking the whole object initialization as if it was only one line.

The [documentation](../blob/master/documentation/SA1118.md) does not cover this case or nested method calls.

**Comment by sharwell** [comment]
Considering that object initializers would need to be indented when they span multiple lines, accepting them as arguments should work well with our existing handling of parameters.
